### PR TITLE
Fix the NSI descriptor

### PR DESCRIPTION
**Pull request by letiemble**
Change some stuff to match the binary distribution.

### CODE REVIEW

File: etc/windows/SyncThingSetup.nsi [hunk 1]
```diff
@@ -71,7 +71,7 @@ Section "Command Line Interface" SecSyncThingCLI
 
   SetOutPath "$INSTDIR"
   
-  File /r "${SOURCEPATH}\syncthing-cli.exe"
```
Comment (calmh) [inline etc/windows/SyncThingSetup.nsi]: syncthing-cli is not in fact the same thing as syncthing so I'm unsure what's the intention here. Your change may very well be valid - but are you sure?

File: etc/windows/SyncThingSetup.nsi [hunk 2]
```diff
@@ -113,4 +113,4 @@ Section "Uninstall"
 
   DeleteRegKey /ifempty HKCU "Software\SyncThing"
```
Comment (calmh) [inline etc/windows/SyncThingSetup.nsi]: Ugh. Can we change this to `Software\Syncthing` as well?

File: etc/windows/SyncThingSetup.nsi (hunk 1 above)
Comment (canton7) [inline etc/windows/SyncThingSetup.nsi]: Yeah it should be `syncthing-cli.exe`: it's in an optional section called "Command Line Interface"

Comment (letiemble) [inline etc/windows/SyncThingSetup.nsi]: Ok. So maybe the README needs to be updated to mention that.

**Comment by canton7** [comment]
To be honest, I have my doubts as to whether this belongs in `/etc` at all. It's part - but not all - of the setup needed to create an installer for syncthing. It's not plug-and-play like everything else in /etc: someone who knows vaguely what they're doing has to set it up, the build it, and finally run it. For an end-user, that's more effort than just following the steps manually.

It's a bit like shipping PKGBUILDs or whatever configuration file is used to build .dep packages.

It would make more sense for a central authority to build the installer and then distribute it. This would put in on the same footing and in the same category as deb packages, dmg images, the other windows installers, etc.

**Comment by calmh** [comment]
I mostly agree with canton7 here, I'm not entirely sure why this is in here to begin with. If nothing else this could fit better as a separate syncthing-windows-installer repo, if that isn't already what Synctrayzor etc is...

**Comment by canton7** [comment]
There are multiple legitimate ways of installing Syncthing on Windows. SyncTrayzor chooses to start on login, whereas this installer installs as a service (I think). I would have to check that it doesn't run as local admin or anything silly before feeling we could officially sanction it though...

**Comment by calmh** [comment]
Right. So I think this doesn't really fit in `etc`, especially as that means it gets shipped in all UNIX packagings and not Windows. If this is actually in use, you're welcome to clone it to a separate repository. :)

**Comment by AudriusButkevicius** [comment]
So that's not what this PR is about, not sure why are we even discussing this here.

**Comment by canton7** [comment]
Fair point

**Comment by calmh** [comment]
This PR raised the point by making me aware of this stuffs existence again. This is no way a bad reflection on @letiemble - I'm very thankful for you noticing a problem here and putting in the time to rectify it for us! But the fact remains that this is code that does something I don't fully understand and isn't tested at all on any regular basis, yet I am supposedly maintaining it. It needs to go.

**Comment by calmh** [comment]
Actually, here's what I'm going to do. I'm going to extract this into a separate repo, as it's no more unmaintained than my small Mac app wrapper script which lives like that, and it obviously holds some value. That repo can then attract maintainers as needed. :)

**Comment by calmh** [comment]
@letiemble https://github.com/syncthing/syncthing-nsi <- could you please refile without the syncthing-cli change, and with a possible not on that dependency? Sorry for all the confusion here.

**Comment by letiemble** [comment]
@calmh No problem. I was evaluating Syncthing on Windows and I blindly followed the instructions listed in /etc/windows/ReadMe.txt so I wasn't aware of the CLI executable. Maybe I should wait for the separate repo before submitting any new PR, shouldn't I ?

**Comment by canton7** [comment]
[The repo already exists](https://github.com/syncthing/syncthing-nsi) :) 

If you're evaluating Syncthing for Windows, take a look at [SyncTrayzor](https://github.com/canton7/SyncTrayzor) :wink:

**Pull request closed without merging**